xbmc/pvr/channels/PVRChannelGroupInternal.cpp

```diff
@@ -282,10 +282,28 @@ std::vector<std::shared_ptr<CPVRChannel>> CPVRChannelGroupInternal::RemoveDelete
 {
   std::vector<std::shared_ptr<CPVRChannel>> removedChannels = CPVRChannelGroup::RemoveDeletedChannels(channels);
 
-  for (const auto& channel : removedChannels)
+  bool channelsDeleted = false;
+
+  const std::shared_ptr<CPVRDatabase> database(CServiceBroker::GetPVRManager().GetTVDatabase());
```

Comment:
use default ctor + assign intsead of copy ctor.
=> const std::shared_ptr<CPVRDatabase> database = CServiceBroker::GetPVRManager().GetTVDatabase();

Reply:
Done